src/Model/common.hpp

```diff
@@ -114,65 +122,121 @@ void seissol::model::getTransposedElasticCoefficientMatrix( seissol::model::Elas
   }
 }
 
-template<typename Tloc, typename Tneigh>
-void seissol::model::getTransposedElasticGodunovState( Material const&                      local,
-                                                       Material const&                      neighbor,
-                                                       enum ::faceType                      faceType,
-                                                       Tloc&                                QgodLocal,
-                                                       Tneigh&                              QgodNeighbor )
+template<typename T>
+void seissol::model::getTransposedBoundaryGodunovState( T&                         QgodLocal,
+                                                        T&                         QgodNeighbor,
+                                                        Eigen::Matrix<real, 9, 9>& R)
 {
-  QgodNeighbor.setZero();
-  
-  real cpL = sqrt((local.lambda + 2.0 * local.mu)       / local.rho);
-  real cpN = sqrt((neighbor.lambda + 2.0 * neighbor.mu) / neighbor.rho);
-  real csL = sqrt(local.mu / local.rho);
-  real csN = sqrt(neighbor.mu / neighbor.rho);
-  
-  real constP = cpN * (local.lambda + 2.0 * local.mu) + cpL * (neighbor.lambda + 2.0 * neighbor.mu);
-  real constS = csN * local.mu + csL * neighbor.mu;
-  
-  QgodNeighbor(0,0) = cpN * (local.lambda + 2.0 * local.mu) / constP;
-  QgodNeighbor(6,0) = (local.lambda + 2.0 * local.mu) * (neighbor.lambda + 2.0 * neighbor.mu) / constP;
-  QgodNeighbor(0,1) = cpN * local.lambda / constP;
-  QgodNeighbor(6,1) = local.lambda * (neighbor.lambda + 2.0 * neighbor.mu) / constP;
-  QgodNeighbor(0,2) = QgodNeighbor(0,1);
-  QgodNeighbor(6,2) = QgodNeighbor(6,1);
-  QgodNeighbor(3,3) = csN * local.mu / constS;
-  QgodNeighbor(7,3) = local.mu * neighbor.mu / constS;
-  QgodNeighbor(5,5) = QgodNeighbor(3,3);
-  QgodNeighbor(8,5) = QgodNeighbor(7,3);
-  QgodNeighbor(0,6) = cpL * cpN / constP;
-  QgodNeighbor(6,6) = cpL * (neighbor.lambda + 2.0 * neighbor.mu) / constP;
-  QgodNeighbor(3,7) = csL * csN / constS;
-  QgodNeighbor(7,7) = csL * neighbor.mu / constS;
-  QgodNeighbor(5,8) = QgodNeighbor(3,7);
-  QgodNeighbor(8,8) = QgodNeighbor(7,7);
-
-  // QgodLocal = I - QgodNeighbor
-  for (unsigned i = 0; i < 9; ++i) {
-    for (unsigned j = 0; j < 9; ++j) {
-      QgodLocal(i,j) = -QgodNeighbor(i,j);
+  for (int i = 0; i < 9; i++) {
+    for (int j = 0; j < 9; j++) {
+      QgodNeighbor(i,j) = std::numeric_limits<double>::signaling_NaN();
     }
-  }  
-  for (unsigned idx = 0; idx < 9; ++idx) {
-    QgodLocal(idx,idx) += 1.0;
   }
-  
-  applyBoundaryConditionToElasticFluxSolver(faceType, QgodNeighbor);
+
+  QgodLocal.setZero();
+  std::array<std::array<int, 2>, 3> traction_indices = {{{0,0}, {1,3}, {2,5}}};
+  std::array<std::array<int, 2>, 3> velocity_indices = {{{0,6}, {1,7}, {2,8}}};
+  //Eigen does not have fancy slicing options :(
```

Comment:
?? https://eigen.tuxfamily.org/dox-devel/group__TutorialSlicingIndexing.html

Reply:
See "Array of indices"

Reply:
Thanks for pointing it out. Apparently the git mirror provides version 3.3.90 although the eigen webpage states, that 3.3.7 is the latest release.